pkg/apis/agent/v1alpha1/validations.go

```diff
@@ -104,10 +111,117 @@ func checkSingleConfigSource(b *Agent) field.ErrorList {
 	return nil
 }
 
-func checkSpec(b *Agent) field.ErrorList {
-	if (b.Spec.DaemonSet == nil && b.Spec.Deployment == nil) || (b.Spec.DaemonSet != nil && b.Spec.Deployment != nil) {
+func checkSpec(a *Agent) field.ErrorList {
+	if (a.Spec.DaemonSet == nil && a.Spec.Deployment == nil) || (a.Spec.DaemonSet != nil && a.Spec.Deployment != nil) {
 		return field.ErrorList{
-			field.Invalid(field.NewPath("spec"), b.Spec, "either daemonset or deployment must be specified"),
+			field.Invalid(field.NewPath("spec"), a.Spec, "either daemonset or deployment must be specified"),
+		}
+	}
+	return nil
+}
+
+func checkEmptyConfigForFleetMode(a *Agent) field.ErrorList {
+	var errors field.ErrorList
+	if a.Spec.FleetModeEnabled() {
+		if a.Spec.Config != nil {
+			errors = append(errors, field.Invalid(
+				field.NewPath("spec").Child("config"),
+				a.Spec.Config,
+				"remove config, it can't be set in fleet mode",
+			))
+		}
+
+		if a.Spec.ConfigRef != nil {
+			errors = append(errors, field.Invalid(
+				field.NewPath("spec").Child("configRef"),
+				a.Spec.ConfigRef,
+				"remove configRef, it can't be set in fleet mode",
+			))
```

Comment:
Elastic Agent config file is empty in case of fleet mode, there is nothing to override. I think it would be misguiding to allow specifying any config when it's not even read by anything.